docs/release-notes/release-notes-0.9.9.md

```diff
@@ -0,0 +1,155 @@
+Joinmarket-clientserver 0.9.9:
+=================
+
+<https://github.com/joinmarket-org/joinmarket-clientserver/releases/tag/v0.9.9>
+
+Please report bugs using the issue tracker at github:
+
+<https://github.com/joinmarket-org/joinmarket-clientserver/issues>
+
+*Users are strongly encouraged to update immediately, mainly because of what is described in the first section below, on "Onion messaging related fixes and changes".*
+
+Upgrading
+=========
+
+To upgrade:
+
+*Reminder: always back up and recreate your joinmarket.cfg file when doing the upgrade; this is to make sure you have the new default settings.*
+
+(If you are upgrading from a version pre-0.7.0 please read the "Upgrading" section in [the 0.7.0 release notes](https://github.com/JoinMarket-Org/joinmarket-clientserver/blob/master/docs/release-notes/release-notes-0.7.0.md).)
+
+(If you are upgrading from a version pre-0.9.0 please read the [release notes](https://github.com/JoinMarket-Org/joinmarket-clientserver/blob/master/docs/release-notes/release-notes-0.9.0.md) to find out how about [fidelity bonds](../fidelity-bonds.md) are relevant to your use-case).
+
+First run the `install.sh` script as mentioned in the README. When prompted to overwrite the directory `jmvenv`, accept.
+
+Note that `./install.sh -?` will show the options for installation. See the section on Installation below for a new option for Tor.
+
+Changes
+===============
+
+### Onion messaging related fixes and changes
+
+#### Changes to directory node defaults
+
+We have removed one of the directory nodes that was proving unresponsive, and added a new one. These of course are only defaults; this list may change quite often (more often than the IRC list used to, probably!). We will investigate ways to let users find other directory nodes than just this list, in future.
+
+We also increase the timeout for making the initial connection, as onion service connections can sometimes be very slow.
+
+Note that this is also related to the following topic (slowing down connection reattempts).
+
+`8ab2d50` Increase directory node connection timeout to 60s
+`ebdbac7` Update configure.py
+`7aebed7` Remove jmdirjmioywe2s5jad7ts6kgcqg66rj6wujj6q77n6wbdrgocqwexzid.onion directory node from default config
+
+#### Slow down connection reattempts to directory nodes
+
+Several users reported not only a failure to connect to directory nodes, but also a very rapid cycling, attempting to reconnect every couple of seconds. Here we fix that bug, and ensure that connection attempts back off exponentially, eventually slowing down to once per ~ 3 hours. This will clear up spam from the log, but also may help by not triggering DOS limits in Tor (further investigation into this issue will be required), once all users update to 0.9.9. **This is one of the main reason users are encouraged to update immediately**.
```

Comment:
> but also may help by not triggering DOS limits in Tor (further investigation into this issue will be required)

I'll just reiterate that, strictly speaking, we don't know that (re: DOS limits in Tor). The way the text is framed sounds to me like "we know the problem is Tor, but further investigation is required". When I think it should sound more like "we don't know what the problem is exactly, it might be Tor limits, it might be Twisted limits, etc., and further investigation is required."

I don't mind that much either way, so I'm okay to leave it as is if you are confident enough Tor is the culprit.

Reply:
I agree to change the text to not *suggest* we know for sure where the problem is.

Reply:
Done in https://github.com/JoinMarket-Org/joinmarket-clientserver/pull/1443/commits/978263b4e72edb18c47c164ce609baa63932da39